coderd/httpmw/patternmatcher/routepatterns_test.go

```diff
@@ -0,0 +1,99 @@
+package patternmatcher_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/coder/coder/coderd/httpmw/patternmatcher"
+)
+
+func Test_RoutePatterns(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name        string
+		patterns    []string
+		errContains string
+		output      string
+	}{
```

Comment:
You may want to include more cases:
* `/`
* `//`
* `/api/`
* /api//v1`

I know that patterns are hard coded, but if we decide to generate them dynamically, some bugs in the future might be harder to investigate.